dev/PushNotifications/PushNotificationsLongRunningTask/ForegroundSinkManager.cpp

```diff
@@ -0,0 +1,36 @@
+﻿#pragma once
+
+#include "pch.h"
+
+#include <NotificationsLongRunningProcess_h.h>
+#include <ForegroundSinkManager.h>
+
+void ForegroundSinkManager::Add(std::wstring const& processName, IWpnForegroundSink* const& sink)
+{
+    auto lock = m_lock.lock_exclusive();
+    m_foregroundMap[processName] = sink;
+}
+
+void ForegroundSinkManager::Remove(std::wstring const& processName)
+{
+    auto lock = m_lock.lock_exclusive();
+    m_foregroundMap.erase(processName);
+}
+
+bool ForegroundSinkManager::InvokeForegroundHandlers(std::wstring const& processName, winrt::com_array<uint8_t> const& payload, ULONG const& payloadSize)
```

Comment:
Who calls this method? Is there a pipe from the LongRunningSink that will be calling into this as a seperate PR?